.github/workflows/nx-dependency-update.yaml

```diff
@@ -25,21 +28,25 @@ jobs:
         id: dependabot-metadata
         uses: dependabot/fetch-metadata@v2
 
+      - name: Setup pnpm
+        uses: pnpm/action-setup@v4
+
       - name: Setup Node.js
         uses: actions/setup-node@v4
         with:
-          cache: npm
+          cache: pnpm
+          node-version: ${{ env.NODE_VERSION }}
 
       - name: Install Dependencies
-        run: npm ci
+        run: pnpm install --frozen-lockfile
 
       - name: Migrate nx
         run: |
-          npx nx migrate ${{ steps.dependabot-metadata.outputs.new-version }} \
+          pnpm exec nx migrate ${{ steps.dependabot-metadata.outputs.new-version }} \
             --from nx@${{ steps.dependabot-metadata.outputs.previous-version }}
 
       - name: Install updated depenencies
-        run: npm install
+        run: pnpm install
```

Comment:
Please add `--frozen-lockfile false`, otherwise this will fail in the CI.

See https://pnpm.io/cli/install#--frozen-lockfile